### PR TITLE
[GH-588] Adjusted oracle transactions 

### LINKED ISSUE #588: Adjust oracle transactions

### PR DESCRIPTION
**Pull request by DanielaIvanova**
resolves #588

### CODE REVIEW

**Comment by 0legIv** [comment]
There is an error in Travis, probably when nodes are synced and you try to add tx to the pool:

`06:47:55.818 [error] 06:47:55.814 [error] Elixir.Aecore.Tx.Pool.Worker: Transaction invalid - Elixir.Aecore.Oracle.Tx.OracleResponseTx: Wrong query_id size: %Aecore.Tx.SignedTx{data: %Aecore.Tx.DataTx{fee: 5, nonce: 3, payload: %Aecore.Oracle.Tx.OracleResponseTx{query_id: <<23, 21, 219, 114, 106, 124, 89, 191, 10, 117, 216, 48, 101, 87, 69, 14, 67, 5, 131, 32, 5, 34, 50, 98, 5, 201, 226, 65, 130, 17, 134, 95, 54, 81, 160, 210, 187, 36, 125, 32, 178, 185, 91, 133, 41, ...>>, response: "boolean"}, senders: [%Aecore.Chain.Identifier{type: :oracle, value: <<23, 21, 219, 114, 106, 124, 89, 191, 10, 117, 216, 48, 101, 87, 69, 14, 67, 5, 131, 32, 5, 34, 50, 98, 5, 201, 226, 65, 130, 17, 134, 95>>}], ttl: 0, type: Aecore.Oracle.Tx.OracleResponseTx}, signatures: [<<106, 133, 204, 176, 159, 115, 181, 56, 246, 1, 125, 172, 250, 204, 144, 5, 72, 240, 152, 205, 95, 15, 124, 40, 198, 149, 249, 83, 128, 184, 12, 73, 179, 105, 77, 76, 12, 118, 177, 46, 20, 193, 211, 238, 184, 138, 219, ...>>]}`

And the TestFramework didn't catch it because it doesn't checks the pool, So I will make an issue to adjust it.

File: apps/aecore/lib/aecore/oracle/oracle.ex [hunk 1]
```diff
@@ -123,44 +123,42 @@ defmodule Aecore.Oracle.Oracle do
...
+      type: OracleResponseTx
+    }
 
     {:ok, tx} = SignedTx.sign_tx(tx_data, pubkey, privkey)
     Pool.add_transaction(tx)
   end
 
-  @spec extend(non_neg_integer(), non_neg_integer(), non_neg_integer()) :: :ok | :error
+  @spec extend(ttl(), non_neg_integer(), non_neg_integer()) :: :ok | :error
   def extend(ttl, fee, tx_ttl \\ 0) do
```
Comment (d-velev) [inline apps/aecore/lib/aecore/oracle/oracle.ex]: In the case that we change it to take this ttl structure then there should be a check if it's relative in validate()

Comment (DanielaIvanova) [reply apps/aecore/lib/aecore/oracle/oracle.ex]: As I understand from `epoch` the ttl for `ExtendTX` can be only `:relative`. Is this correct?

Comment (d-velev) [reply apps/aecore/lib/aecore/oracle/oracle.ex]: for some reason yes

**Pull request merged**